src/dopri.c

```diff
@@ -742,7 +742,9 @@ const double* dopri_find_time(dopri_data *obj, double t) {
     const double
       t0 = ((double*) ring_buffer_tail(obj->history))[idx_t],
       t1 = ((double*) ring_buffer_tail_offset(obj->history, n - 1))[idx_t];
-    idx0 = min_size((t - t0) / (t1 - t0) / (n - 1), n - 1);
+    if ((t0 - t) * (t1 - t) < 0) {
+      idx0 = (t - t0) / (t1 - t0) / (n - 1);
+    }
```

Comment:
So as we discussed: the values of t0, t1 and t (11, 12, 0.02) that caused the crash are perfectly reasonable; this part of the code should accept them gracefully, and then the code following this uses idx as a start point for searching the buffer. 

Hence, zero is the best answer for idx to have after this bit of code, in the cases where it would previously have been negative.